Makefile

```diff
@@ -0,0 +1,41 @@
+
```

Comment:
the following makefile helps to perform different tasks on the project without forcing you to keep their syntax in mind. For example, to create a new migration, you write: `make create-migration`, it promts for its name and creates it as well. Most of modern shells show possible make targets when pressing tab key after `make ...`.

Reply:
How about just using lein aliases so that we avoid new tools and have more power by being able to access our project and language in lein in case we need it?

Reply:
Well, please do not take me wrong, I'm against lein aliases. There are some advantages of makefile over lein that I'm going to share.

First, starring any lein command takes a REALLY long time. Even without auto-tasks, the delay might be about 5 to 10 seconds. Now Imagine you made a misprint and should call it and wait again.

Then, with auto-tasks specified in lein project, the process of starting repl takes up to minute, because of compiling contracts to Java classes.

Finally, lein utility was created for managing clojure-driven project, but not a common project. For example, in make file you may declare dependencies, saying something like `make uberjar` that automatically calls building less, js, contracts, etc tracking their success status.

Makefile is rather a collection of actions that might be performed over the project in general. Say, compiling CSS files is not about clojure, it is more general part. This file is easier to read because of clear declarative syntax rather then lein project file that usually brings lots of additional stuff.

But still, if you guys do not want to have makefile in the project, there no any claims from my side. I'll just keep it locally.

Reply:
sorry, a funny misprint: I'm **not** against aliases

Reply:
@igrishaev 

Thanks for the explanations! :)

I'm okay with the makefile as long as it is just to run commands in an easier way, not to develop more logics there.

As for the time that lein takes to run commands and the management those commands, we could probably overcome that with [lumo](https://github.com/anmonteiro/lumo) and/or [boot](https://github.com/boot-clj/boot) to continue having the power of clojure and the integration with the whole project, but that is probably not a priority right now.